students/johnn/mailroom/tests/test_mailroom.py

```diff
@@ -1,32 +1,125 @@
 #!/usr/bin/env python3
 
-import pytest
+"""
+test script for mailroom
+"""
+
 import io
 import os
-from pprint import pprint
+from unittest.mock import (patch, MagicMock)
+import pytest
 from mailroom import security
 from mailroom.donors import Donor, Donors, load_donor_file, save_donor_file
-from mailroom.ui import print_thank_you, print_lines, thank_all_donors, list_donors
+from mailroom.ui import (print_thank_you,
+                         print_lines,
+                         thank_all_donors,
+                         list_donors,
+                         safe_input,
+                         user_login,
+                         user_logout,
+                         get_donation_amount,
+                         data_entry,
+                         main)
+
+
+# define a default user so we can make objects
+security.user = "Test User"
+
+
+def test_user_logout():
+    """ test the logout routine """
+    assert security.user == "Test User"
+    user_logout()
+    assert security.user is None
+
+
+# define a default user so we can make objects
+security.user = "Test User"
+
```

Comment:
Since tests can be run in any order, you really have no idea what affect assigning security.user in the file global space will have on your tests. This would be a good candidate for a fixture.